packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-oidc.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useCallback } from 'react';
+import {
+  Accordion,
+  Checkbox,
+  Description,
+  FormFieldContainer,
+  Label,
+  TextInput,
+} from '@mongodb-js/compass-components';
+import type ConnectionStringUrl from 'mongodb-connection-string-url';
+import type { ConnectionOptions } from 'mongodb-data-service';
+import type { AuthFlowType } from '@mongodb-js/oidc-plugin';
+
+import type { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+import type { ConnectionFormError } from '../../../utils/validation';
+import { errorMessageByFieldName } from '../../../utils/validation';
+import {
+  getConnectionStringUsername,
+  parseAuthMechanismProperties,
+} from '../../../utils/connection-string-helpers';
+
+type OIDCOptions = NonNullable<ConnectionOptions['oidc']>;
+
+function AuthenticationOIDC({
+  connectionStringUrl,
+  updateConnectionFormField,
+  errors,
+  connectionOptions,
+}: {
+  connectionStringUrl: ConnectionStringUrl;
+  errors: ConnectionFormError[];
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionOptions: ConnectionOptions;
+}): React.ReactElement {
+  const username = getConnectionStringUsername(connectionStringUrl);
+  const usernameError = errorMessageByFieldName(errors, 'username');
+
+  const handleFieldChanged = useCallback(
+    (key: keyof OIDCOptions, value?: OIDCOptions[keyof OIDCOptions]) => {
+      return updateConnectionFormField({
+        type: 'update-oidc-param',
+        key: key,
+        value,
+      });
+    },
+    [updateConnectionFormField]
+  );
+
+  const authMechanismProperties =
+    parseAuthMechanismProperties(connectionStringUrl);
+  const allowedHosts = authMechanismProperties.get('ALLOWED_HOSTS');
+  const hasEnabledUntrustedEndpoints = allowedHosts === '*';
```

Comment:
Wasn’t quite sure whether to comment this here or in `connection-string-helpers.ts`, but `ALLOWED_HOSTS` is not an option that can be set in the URL, it needs to be set in the driver `authMechanismProperties` struct.

It might be easier to extends `ConnectionOptions['oidc']` with a boolean option here, e.g. `oidc?: OIDCOptions & { enableUntrustedEndpoints?: boolean }`; or similar.

Reply:
I've updated it to use a `enableUntrustedEndpoints` on the `OIDCOptions` which is then resolved before connecting in `data-service`. Was that similar to what you had in mind? Haven't added a test yet.

Reply:
> Was that similar to what you had in mind?

Yes :)

> Would it work/be cleaner nicer to add the `authMechanismProperties` to the `ConnectionOptions` and then when we connect we pull it in?

Yeah, I think on the one hand this is probably a bit closer to the spirit of `ConnectionOptions`, on the other hand, we really don’t want to add more usage of `authMechanismProperties` outside of the connection string. If we do add something to `ConnectionOptions`, it would probably be more about including all possible `DevtoolsConnectOptions` in it, so that we can handle e.g. `ConnectionFleOptions` there as well.

Reply:
Sounds good, let's keep it as is with the `enableUntrustedEndpoints` and we can looks at `DevtoolsConnectOptions` down the line.